### PR TITLE
Script for publishing code documentation

### LINKED ISSUE #694: Generate code documentation with ApiGen
#### Expected behavior:
Use ApiGen to create the code documentation of the project

 Reference:

* [ApiGen](https://github.com/ApiGen/ApiGen)

### PR DESCRIPTION
**Pull request by DIOHz0r**
Signed-off-by: Domingo Oropeza <doropeza@teclib.com>

### Changes description

This PR update the CI scripts for create code documentation

### Checklist

Please check if your PR fulfills the following specifications:

- [ ] Tests for the changes have been added
- [ ] Docs have been added/updated

### Estimated time

|Assignee|:tomato:|
|:---|:---:|
|@DIOHz0r |8|

### References

Closes #694

### CODE REVIEW

File: tests/apidocs.sh [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            git add development/code-documentation/"$TRAVIS_BRANCH"/*
+
+            # upload_files
+            echo "pushing the up to date documents"
+            git commit --message "docs: update code documentation report"
+            git fetch origin-pages
+            git rebase origin-pages/gh-pages
+            git push --quiet --set-upstream origin-pages gh-pages --force
+        else
+           echo "ApiGen not found, see http://www.apigen.org/"
```
Comment (btry) [inline tests/apidocs.sh]: Maybe return an error code to ensure the build fails ?

File: tests/apidocs.sh [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+#
+# Apidoc doesn't work on php 7.2 because compatibility problems,
+# that's why this script is used on php 7.1 until the problem is solved
+#
+
+# find if we are in a valid branch to build docs
+if echo "$TRAVIS_BRANCH" | grep -q -P '^(master|develop|support/|release/)'; then
+    REGEX_BRANCH=true
+else
+    REGEX_BRANCH=false
```
Comment (btry) [inline tests/apidocs.sh]: This variable has a bad name. Or tells of we must run or not apigen.

BUILD_DOC_API is a better name

**Comment by btry** [comment]
I need to understand why you don't build the doc with php 7.2. Is it to solve the dependencies conflict you talked about ?

File: tests/apidocs.sh (hunk 1 above)
Comment (DIOHz0r) [reply tests/apidocs.sh]: this code is triggered after one success build, is that possible?

File: tests/apidocs.sh (hunk 2 above)
Comment (DIOHz0r) [reply tests/apidocs.sh]: I left the same code name because it's the same conditional statement used on the after_success script.

Comment (btry) [reply tests/apidocs.sh]: then my comment apply to the other script.

**Pull request merged**